azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKInitialization.kt

```diff
@@ -112,9 +137,92 @@ internal class CallingSDKInitialization(
     }
 
     fun dispose() {
+        incomingCall?.removeOnCallEndedListener(onIncomingCallEnded)
+        incomingCall = null
+        callAgentCompletableFuture?.get()?.removeOnIncomingCallListener(callCompositeIncomingCallListener)
         callAgentCompletableFuture?.get()?.dispose()
         callAgentCompletableFuture = null
         callClientInternal = null
         callClientCompletableFuture = null
     }
+
+    fun rejectIncomingCall(callID: String): CompletableFuture<Void> {
+        val completableFuture: CompletableFuture<Void> = CompletableFuture<Void>()
+        if (incomingCall == null || incomingCall?.id != callID) {
+            completableFuture.completeExceptionally(IllegalStateException("No incoming call to reject"))
+        }
+        incomingCall?.reject()?.whenComplete { result, exception ->
+            if (exception != null) {
+                completableFuture.completeExceptionally(exception)
+            }
+            incomingCall = null
+            completableFuture.complete(result)
+        }
+        return completableFuture
+    }
+
+    fun handlePushNotification(pushNotification: CallCompositePushNotification): CompletableFuture<Void> {
+        val completableFuture: CompletableFuture<Void> = CompletableFuture<Void>()
+        if (pushNotification.notificationInfo == null) {
+            completableFuture.completeExceptionally(IllegalArgumentException("Push notification info is null"))
+        }
+        if (isAnyCallActive()) {
+            completableFuture.completeExceptionally(IllegalStateException("Currently UI is busy with an active call - only one call is supported"))
+        }
+        createCallAgent().whenComplete { callAgent, callAgentError ->
+            if (callAgentError != null) {
+                completableFuture.completeExceptionally(callAgentError)
+            }
+            val pushNotificationInfo = PushNotificationInfo.fromMap(pushNotification.notificationInfo)
+            callAgent?.handlePushNotification(pushNotificationInfo)?.whenComplete { result, exception ->
+                if (exception != null) {
+                    completableFuture.completeExceptionally(exception)
+                }
+                completableFuture.complete(result)
+            }
+        }
+        return completableFuture
+    }
+
+    private fun onIncomingCall(incomingCall: IncomingCall?) {
+        if (isAnyCallActive()) {
+            logger.info("Currently UI Library is busy with an active call - only one call is supported")
+            return
+        }
+        incomingCall?.let {
+            this.incomingCall = incomingCall
+            it.addOnCallEndedListener(onIncomingCallEnded)
+            notifyHandlerForIncomingCall(it)
+        }
+    }
+
+    private fun notifyHandlerForIncomingCall(incomingCall: IncomingCall) {
+        callCompositeConfiguration.callCompositeEventsHandler.getOnIncomingCallHandlers().forEach {
+            it.handle(
```

Comment:
it.handle(...) needs be wrapped in try catch to avoid crush if Contoso's code throws an exception.